src/main/java/algorithms/curated170/medium/InorderSuccessorInBST2.java

```diff
@@ -0,0 +1,47 @@
+package algorithms.curated170.medium;
+
+public class InorderSuccessorInBST2 {
+
+    public Node sucNode(Node start) {
+        if (start == null) {
+            return null;
+        }
+        if (start.right == null) {
+            while (start.parent != null) {
+                if (start.parent.left == start) {
+                    return start.parent;
+                }
+                start = start.parent;
+            }
+            return null;
```

Comment:
We can combine if and while like this:

```
  public Node inorderSuccessor(Node start) {
        if (start == null) {
            return null;
        }
      if (start.right != null) return findSuc(start.right);
       
      while (start.parent != null && start.parent.left != start) {
                
                start = start.parent;
       }
      
       return start.parent;
     
        
    }
```


![image](https://user-images.githubusercontent.com/60903744/117544210-13f67a00-b029-11eb-8cc9-c24dce0e5a5d.png)